(fix) prevent 500 server error on a just removed folder when listing files

tobitege opened the pull request:
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

If a folder in the workspace is being removed (in the actual filesystem, outside of OpenHands) and then in the UI the "Refresh" icon on the file explorer is clicked, an error "500 internal server error" will appear in the UI. This PR prevents this.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Instead of returning a 400 error from the runtime client back in case a folder no longer exists, this should just return an empty list and not error about it, which is confusing in the UI.

---
**Other references**

neubig reviewed (commented):
I'm a little bit confused by this. Is it only the case that this error would be thrown when a user just removed a folder? It seems like it also might be thrown if we call `list_files` on a file/directory that never existed in the first place.

tobitege commented:
> I'm a little bit confused by this. Is it only the case that this error would be thrown when a user just removed a folder? It seems like it also might be thrown if we call `list_files` on a file/directory that never existed in the first place.

Just for illustration, my workspace has these entries:
![grafik](https://github.com/user-attachments/assets/509a1de8-658b-4728-a7bb-ecc77096c48e)

So if I'd go outside of OH and removed the folder "temp2" myself and then clicked the "Refresh" button, the UI would retrieve the error as the `list_files` would still think it existed and try to get the content for it, still. 
And that error message is now removed.

xingyaoww reviewed (approved):
This LGTM! Although i wonder if the file structure will also be updated on the FE side, so it will not call list file on the same folder next time?

tobitege commented:
> This LGTM! Although i wonder if the file structure will also be updated on the FE side, so it will not call list file on the same folder next time?

Yes, the file explorer is updated correctly.
